src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -196,10 +196,6 @@ protected AbortException abortUnrecognizedArgument(String argument) {
     }
 
     private static int runRubyMain(Context.Builder contextBuilder, CommandLineOptions config) {
-        if (config.getOption(OptionsCatalog.EXECUTION_ACTION) == ExecutionAction.UNSET) {
-            config.getOption(OptionsCatalog.DEFAULT_EXECUTION_ACTION).applyTo(config);
-        }
-
         if (config.getOption(OptionsCatalog.EXECUTION_ACTION) == ExecutionAction.NONE) {
             return 0;
         }
```

Comment:
Will this still catch just printing the version and not create a RubyContext in that case?
Or could UNSET resolve to NONE?

Reply:
yes, this check is done after the version is printed.
When somebody changes the default_execution_action to none it could go to context and do nothing, which will not happen normally when using launcher. Safe to ignore.